projects/plugins/crm/src/automation/workflow/class-workflow-repository.php

```diff
@@ -117,6 +114,15 @@ public function find_by( array $criteria ): array {
 			$query .= ' WHERE ' . implode( ' AND ', $where );
 		}
 
+		// Build pagination condition.
+		if ( isset( $criteria['page'] ) || isset( $criteria['per_page'] ) || isset( $criteria['offset'] ) ) {
```

Comment:
Better if we add the pagination/order functionality as params in the method:

`public function find_by( array $criteria, $order_by = 'id', int $limit = null, int $offset = 0 ): array {`

I prefer the `limit` and `offset` terms to match with the SQL clauses.

The reason is that we could have columns with those names: `page`, `per_page`, and `offset`.

By the way, `offset` and `page` would be the same.

Reply:
We can definitely change the naming inside the repository itself.
The reason I went with that naming scheme is due to the REST API docs to follow best practice, but those can still be upheld in the endpoint controller instead: https://developer.wordpress.org/rest-api/using-the-rest-api/pagination/#pagination-parameters

I'll give it a look shortly

Reply:
>By the way, offset and page would be the same.

Agreed. There's a comment that also explains it for the code itself.

https://github.com/Automattic/jetpack/pull/33296/files#diff-df77d7f24c7769b6989fff4eb24d14d497b61c055a2971875cd46af9d5df6d7cR306-R309